SuperblocksHQ/superblocks-lab
# Add a warning modal when trying to deploy to Mainnet

### Description of the Change

Simply add a new confirmation modal when trying to deploy a contract to Mainnet. Deploying to Mainnet can be a scary thing (at least for now), so we want to make sure the user is 100% sure of what she/he is doing.

Also by doing so, a few improvements are coming along:
- Extracted all none-ui code outside the component and place it in its own class
- Abstracted the toolbar used for the deployer and compiler into its own component
- Modified the projects Redux store to be able to start storing more useful information about the project among other components
- Abstracted the Networks outside the App component

### Benefits

The main benefit of this is making sure no one accidentally deploy a contract to Mainnet

### Possible Drawbacks

That adds a little bit of friction when deploying, but at least for now, it is the intended behavior.

### Verification Process
1. Open any project or create a new one
2. Make sure you have selected Mainnet as network in Metamask
3. Select the Mainnet network in Lab
4. Deploy a contract. Once this is done the modal should appear
5. Enter the project name in order to deploy to Mainnet
6. Deploy the contract to Mainnet using Metamask

<img width="1398" alt="screenshot 2018-11-15 at 06 38 22" src="https://user-images.githubusercontent.com/7814134/48532534-18e77200-e8a1-11e8-9d90-c58dd216a5dd.png">


### Github Issues
This fixes #162
